journald/lib_journald.cpp

```diff
@@ -166,6 +172,28 @@ class JournaldContainerLoggerProcess :
       }
     }
 
+    foreachpair (const string& key,
+                 const JSON::Value& value,
+                 overriddenFlags.extra_labels.values) {
+      // Skip invalid labels.
+      if (!value.is<JSON::String>()) {
+        continue;
+      }
+
+      string _value = value.as<JSON::String>().value;
+
+      totalSize += LABEL_PADDING_SIZE + key.size() + _value.size();
+
+      // Stop copying arbitrary labels once we hit a size limit.
+      if (totalSize > flags.max_label_payload_size) {
+        break;
+      }
+
+      label.set_key(key);
+      label.set_value(_value);
+      labels.add_labels()->CopyFrom(label);
+    }
+
     // NOTE: This field is required by the master/agent, but the protobuf
     // is optional for backwards compatibility.
     //
```

Comment:
The NOTE right below this line could be updated to match some of the new recovery logic we've added since this comment was added:
```
    // NOTE: It is possible for the ExecutorInfo object to be blank in the
    // following cases:
    //   * Prior to Mesos 1.5.0, the ContainerConfig object is not checkpointed
    //     and hence, any containers recovered from earlier versions will be
    //     missing some metadata.
    //     See: https://issues.apache.org/jira/browse/MESOS-6894
    //   * Standalone containers naturally do not have this field set.
```

Reply:
Done. Thanks!